pkg/controller/factory/base_controller.go

```diff
@@ -202,7 +203,12 @@ func (c *baseController) runWorker(queueCtx context.Context) {
 // reconcile wraps the sync() call and if operator client is set, it handle the degraded condition if sync() returns an error.
 func (c *baseController) reconcile(ctx context.Context, syncCtx SyncContext) error {
 	err := c.sync(ctx, syncCtx)
-	return c.reportDegraded(ctx, err)
+	degradedErr := c.reportDegraded(ctx, err)
+	if apierrors.IsNotFound(degradedErr) && management.IsOperatorRemovable() {
+		// The operator tolerates missing CR, therefore don't report it up.
+		return err
+	}
```

Comment:
I might be a little confused by this change, but shouldn't this check to done before `reportDegrade()`? That function sets `Degraded=true` if `err != nil`, even if `err` is `NotFound`.

Or am I missing something?

Reply:
`c.sync` may return `NotFound` from various reasons. `reportDegraded` returns `NotFound` only when the CR does not exist, therefore I check only that.